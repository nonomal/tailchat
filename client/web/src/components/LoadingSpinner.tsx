import React from 'react';
import { t } from 'tailchat-shared';
import { Spinner } from './Spinner';

interface LoadingSpinnerProps {
  tip?: string;
}
export const LoadingSpinner: React.FC<LoadingSpinnerProps> = React.memo(
  (props) => {
    return (
      <div>
        <Spinner />
        {props.tip ?? t('加载中...')}
      </div>
    );
  }
);
LoadingSpinner.displayName = 'LoadingSpinner';
