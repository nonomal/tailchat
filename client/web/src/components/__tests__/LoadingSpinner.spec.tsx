import { render } from '@testing-library/react';
import React from 'react';
import { LoadingSpinner } from '../LoadingSpinner';

describe('LoadingSpinner', () => {
  test('render', () => {
    const wrapper = render(<LoadingSpinner />);
    expect(wrapper.container).toMatchSnapshot();
  });

  test('render with text', () => {
    const wrapper = render(<LoadingSpinner tip="加载中..." />);
    expect(wrapper.container).toMatchSnapshot();
  });
});
