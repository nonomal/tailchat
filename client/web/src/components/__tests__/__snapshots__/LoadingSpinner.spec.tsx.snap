// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner render 1`] = `
<div>
  <div>
    [iconify icon="mdi:loading"]
    加载中...
  </div>
</div>
`;

exports[`LoadingSpinner render with text 1`] = `
<div>
  <div>
    [iconify icon="mdi:loading"]
    加载中...
  </div>
</div>
`;
